src/Files.App/App.xaml.cs

```diff
@@ -448,19 +449,41 @@ private static void AppUnhandledException(Exception ex, bool shouldShowNotificat
 				{
 					Buttons =
 					{
-						new ToastButton("ExceptionNotificationReportButton".GetLocalizedResource(), "report")
+						new ToastButton("ExceptionNotificationReportButton".GetLocalizedResource(), Constants.GitHub.BugReportUrl)
 						{
-							ActivationType = ToastActivationType.Foreground
+							ActivationType = ToastActivationType.Protocol
 						}
 					}
-				}
+				},
+				ActivationType = ToastActivationType.Protocol
 			};
 
 			// Create the toast notification
 			var toastNotif = new ToastNotification(toastContent.GetXml());
 
 			// And send the notification
 			ToastNotificationManager.CreateToastNotifier().Show(toastNotif);
+
+			// Restart the app
+			var userSettingsService = Ioc.Default.GetRequiredService<IUserSettingsService>();
+			var lastSessionTabList = userSettingsService.PreferencesSettingsService.LastSessionTabList;
+
+			if (userSettingsService.PreferencesSettingsService.LastCrashedTabList?.SequenceEqual(lastSessionTabList) ?? false)
+			{
+				// Avoid infinite restart loop
```

Comment:
Are there any known situations where the app can get into a restart loop?

Reply:
No. Not all crashes are handled with `AppUnhandledException`, but in which case the `LastSessionTabList` is updated to "Home" so that the crashed page is not displayed again.